### PR TITLE
[PREVIEW] SL-1111 delete hearing part

### PR DESCRIPTION
**Pull request by filipczyks**
https://tools.hmcts.net/jira/browse/SL-1111

### CODE REVIEW

File: src/app/hearing-part/components/delete-hearing-part-dialog/delete-hearing-part-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<div id="deleteHearingPartDialog" draggable="true" class="container">
+  Do you want to remove listing request for case number <strong>{{hearingPart.caseNumber}}</strong> ?
```
[the remark this answers is not in the document]
Comment (filipczyks) [reply src/app/hearing-part/components/delete-hearing-part-dialog/delete-hearing-part-dialog.component.html]: done

File: src/app/hearing-part/components/delete-hearing-part-dialog/delete-hearing-part-dialog.component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+export class DeleteHearingPartDialogComponent extends DraggableDialog {
+
+  constructor(
+    public dialogRef: MatDialogRef<any>,
+    @Inject(MAT_DIALOG_DATA) public hearingPart: HearingPartViewModel,
+    private readonly store: Store<State>) {
+    super(dialogRef);
+  }
+
+  onYesClick(hearingPart: HearingPartViewModel) {
```
[the remark this answers is not in the document]
Comment (filipczyks) [reply src/app/hearing-part/components/delete-hearing-part-dialog/delete-hearing-part-dialog.component.ts]: yes, there's no need to pass this from the view, removed

**Comment by satyachundur** [comment]
@filipczyks  -  Unit tests are failing for this.

File: e2e/specs/listing-request.e2e-spec.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { NavigationFlow } from '../flows/navigation.flow'
+import { LoginFlow } from '../flows/login.flow'
+import { SessionSearchPage } from '../pages/session-search.po'
+import { ListingCreationPage } from '../pages/listing-creation.po'
+import { DeleteHearingPartDialogPage } from '../pages/delete-hearing-part-dialog.po'
+
+require('../utils/driver-setup.ts')
```
Comment (zielinskikamil) [inline e2e/specs/listing-request.e2e-spec.ts]: use import instead of require

File: e2e/utils/driver-setup.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { browser, protractor } from 'protractor';
+
+const origFn = browser.driver.controlFlow().execute;
+
+browser.driver.controlFlow().execute = function() {
+  const args = arguments;
+
+  // queue 100ms wait
+  origFn.call(browser.driver.controlFlow(), function() {
```
Comment (zielinskikamil) [inline e2e/utils/driver-setup.ts]: you can move this setup into global `onPrepare` in `protractor.conf.js`

File: src/app/hearing-part/components/delete-hearing-part-dialog/delete-hearing-part-dialog.component.spec.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      component.onYesClick(createHearingPartViewModel());
+      const deleteAction = storeSpy.calls.argsFor(0)[0] as Delete;
+
+      expect(deleteAction.type).toEqual(HearingPartActionTypes.Delete);
+      expect(deleteAction.payload).toEqual(ID);
+    });
+  });
+});
+
+function createHearingPartViewModel() {
```
Comment (zielinskikamil) [inline src/app/hearing-part/components/delete-hearing-part-dialog/delete-hearing-part-dialog.component.spec.ts]: plz add return type

File: src/app/hearing-part/effects/hearing-part.effects.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@ import { catchError, mergeMap } from 'rxjs/operators';
 import { of } from 'rxjs/observable/of';
 import { Action } from '@ngrx/store';
 import {
-    AssignToSession,
+    AssignToSession, DeleteComplete,
```
Comment (zielinskikamil) [inline src/app/hearing-part/effects/hearing-part.effects.ts]: can you move this `DeleteComplete` to the new line?

**Pull request closed without merging**